services/movie_session.py

```diff
@@ -0,0 +1,50 @@
+import datetime
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession, CinemaHall, Movie
+
+
+def create_movie_session(movie_show_time: datetime,
+                         movie_id: int, cinema_hall_id: int) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall=CinemaHall.objects.get(id=cinema_hall_id),
+        movie=Movie.objects.get(id=movie_id)
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet | MovieSession:
+    set_ = MovieSession.objects.all()
+    if session_date:
+        dat = datetime.datetime.strptime(session_date, "%Y-%m-%d")
+        # set_ = set_.filter(show_time__day=session_date.day)
+        # set_ = set_.filter(show_time__month=session_date.month)
+        # set_ = set_.filter(show_time__year=session_date.year)
+        set_ = set_.filter(show_time__date=dat)
+        # set_ = set_.filter(show_time__date=session_date)
+    return set_
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> None:
+    session = MovieSession.objects.get(id=session_id)
+    if show_time:
+        session.show_time = show_time
+    if movie_id:
+        session.movie = Movie.objects.get(id=movie_id)
+    if cinema_hall_id:
+        session.cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
```

Comment:
Same here: Don't load the DB when it's not needed.

Reply:
Same: use .get() is a good practice

Reply:
Disagree with you. Read the checklist (code efficiency): https://github.com/mate-academy/py-movie-moviesession-cinemahall/blob/master/checklist.md#code-efficiency

Reply:
so please, fix it